extensions/azurecore/src/account-provider/auths/azureAuthCodeGrant.ts

```diff
@@ -222,6 +222,13 @@ export class AzureAuthCodeGrant extends AzureAuth {
 		});
 
 		return new Promise<string>((resolve, reject) => {
+			server.on('/redirect', (req, reqUrl, res) => {
+				const state = reqUrl.query.state as string ?? '';
+				const port = state.split(',')[0];
```

Comment:
This may be undefined if there's no , in the string. You should make sure to handle that situation (I assuming in that case you'd just not include the `:${port}` from the URL below)